jolie/src/main/java/jolie/monitoring/events/ProtocolMessageEvent.java

```diff
@@ -0,0 +1,74 @@
+package jolie.monitoring.events;
+
+import jolie.monitoring.MonitoringEvent;
+import jolie.runtime.Value;
+
+public class ProtocolMessageEvent extends MonitoringEvent {
+
+	public enum Protocol {
+		SOAP( "soap" ), HTTP( "http" );
+
+		private String protocol;
+
+		Protocol( String protocol ) {
+			this.protocol = protocol;
+		}
+
+		public String getProtocol() {
+			return protocol;
+		}
+	}
+
+	public static enum FieldNames {
+		PROTOCOL( "protocol" ), MESSAGE( "message" ), HEADER( "header" ), BODY( "body" ), PROCESSID(
+			"processId" ), RAWID( "rawId" );
+
+		private String fieldName;
+
+		FieldNames( String name ) {
+			this.fieldName = name;
+		}
+
+		public String getName() {
```

Comment:
`name()`

Reply:
[ERROR] /mnt/d/my-jolie-fork/jolie/src/main/java/jolie/monitoring/events/ProtocolMessageEvent.java:[54,31] name() in jolie.monitoring.events.ProtocolMessageEvent.Field cannot override name() in java.lang.Enum

choosed to call fieldName

Reply:
Good point. Then maybe `id()`?

Reply:
(PS: the reason is that we do not use the `get` prefix in getters in the Jolie codebase.)

Reply:
enum FieldName changed to enum Field

hidden field named id,
access method to this field changed in id()